### ISSUE TITLE
[Unstable 0.1004.0.0] Mid-round crash when playing singleplayer

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#3161]
Tester reported on Discord that their game keeps constantly crashing in middle of a round, they posted 4 crash report on the baro-unstable channel which all had the exact same cause for the crash. 

https://discordapp.com/channels/103209684680323072/668819422919524352/723881155723657276

No clear reproduction steps to give, tester mentioned that the crash always occurred when they were surrounded by crawlers and they ordered bots to fix/shoot. 

I tested out the same exact scenario but was unable to crash the game.  

```
Barotrauma Client crash report (generated on 18-6-2020 18:59:43)

Barotrauma seems to have crashed. Sorry for the inconvenience! 


CC6A61ED7CA4D1D86D6B0496510A8428


Game version 0.1004.0.0 (UnstableWindows, branch feature/campaign-groundwork, revision 5a89e70b4)
Graphics mode: 1920x1080 (BorderlessWindowed)
VSync ON
Language: English
Selected content packages: Vanilla 0.9
Level seed: Ino LineaAttis
Loaded submarine: Typhon (2BEDE8C973423D67F277B73B09E365F7)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.18362.0 64 bit
    GPU name: NVIDIA GeForce GTX 970
    Display mode: {Width:1920 Height:1080 Format:Color AspectRatio:1,7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Void CreateCommandUI(Barotrauma.Entity, Boolean)
Stack trace: 
   at Barotrauma.CrewManager.CreateCommandUI(Entity entityContext, Boolean forceContextual) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameSession\CrewManager.cs:line 1672
   at Barotrauma.CrewManager.UpdateProjectSpecific(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameSession\CrewManager.cs:line 1260
   at Barotrauma.CrewManager.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\GameSession\CrewManager.cs:line 65
   at Barotrauma.GameMode.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\GameSession\GameModes\GameMode.cs:line 65
   at Barotrauma.CampaignMode.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameSession\GameModes\CampaignMode.cs:line 199
   at Barotrauma.SinglePlayerCampaign.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameSession\GameModes\SinglePlayerCampaign.cs:line 387
   at Barotrauma.GameSession.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\GameSession\GameSession.cs:line 466
   at Barotrauma.GameScreen.Update(Double deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 133
   at Barotrauma.GameMain.Update(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 958
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[18-6-2020 18:43:05] Attempted to access a potentially removed character. Character: Neal Heuer, id: 480, removed: True. AnimController.Collider == null
   at System.Environment.get_StackTrace()
   at Barotrauma.Character.get_SimPosition() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 660
   at Barotrauma.Character.get_Position() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 677
   at Barotrauma.Entity.get_WorldPosition() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Map\Entity.cs:line 102
   at Barotrauma.TriggerAction.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\TriggerAction.cs:line 70
   at Barotrauma.ScriptedEvent.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\ScriptedEvent.cs:line 182
   at Barotrauma.EventManager.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\EventManager.cs:line 510
   at Barotrauma.GameSession.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\GameSession\GameSession.cs:line 465
   at Barotrauma.GameScreen.Update(Double deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 133
   at Barotrauma.GameMain.Update(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 958
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 504
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
[18-6-2020 18:43:05] Attempted to access a potentially removed character. Character: Christiana Avery, id: 476, removed: True. AnimController.Collider == null
   at System.Environment.get_StackTrace()
   at Barotrauma.Character.get_SimPosition() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 660
   at Barotrauma.Character.get_Position() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 677
   at Barotrauma.Entity.get_WorldPosition() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Map\Entity.cs:line 102
   at Barotrauma.TriggerAction.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\TriggerAction.cs:line 70
   at Barotrauma.ScriptedEvent.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\ScriptedEvent.cs:line 182
   at Barotrauma.EventManager.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\EventManager.cs:line 510
   at Barotrauma.GameSession.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\GameSession\GameSession.cs:line 465
   at Barotrauma.GameScreen.Update(Double deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 133
   at Barotrauma.GameMain.Update(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 958
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 504
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
[18-6-2020 18:43:03] Generating an outpost for the start of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 18:43:02] Generating an outpost for the end of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 18:42:52] Transitioning to Ino LineaAttis (current location: Attis City, selected location: Ino Linea Institute of Astrobiology, leaving sub: Typhon, at start: True, at end: False, transition type: LeaveLocation)
[18-6-2020 18:42:19] Generating an outpost for the start of the level... (Location: Attis City, level type: Outpost)
[18-6-2020 18:42:10] Transitioning to Attis (current location: Attis City, selected location: Ino Linea Institute of Astrobiology, leaving sub: Typhon, at start: True, at end: False, transition type: ReturnToPreviousLocation)
[18-6-2020 18:41:33] Generating an outpost for the end of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 18:41:33] Generating an outpost for the start of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 18:19:28] Generating an outpost for the end of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 18:19:28] Generating an outpost for the start of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 18:08:15] Generating an outpost for the end of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 18:08:14] Generating an outpost for the start of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 17:55:01] Generating an outpost for the end of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 17:55:01] Generating an outpost for the start of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 17:35:56] Attempted to access a potentially removed character. Character: Christiana Avery, id: 1406, removed: True. AnimController.Collider == null
   at System.Environment.get_StackTrace()
   at Barotrauma.Character.get_SimPosition() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 660
   at Barotrauma.Character.get_Position() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 677
   at Barotrauma.Entity.get_WorldPosition() in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Map\Entity.cs:line 102
   at Barotrauma.TriggerAction.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\EventActions\TriggerAction.cs:line 70
   at Barotrauma.ScriptedEvent.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\ScriptedEvent.cs:line 182
   at Barotrauma.EventManager.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Events\EventManager.cs:line 510
   at Barotrauma.GameSession.Update(Single deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\GameSession\GameSession.cs:line 465
   at Barotrauma.GameScreen.Update(Double deltaTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 133
   at Barotrauma.GameMain.Update(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 958
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 504
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in C:\Users\Joonas\Documents\Barotrauma-development\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in C:\Users\Joonas\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
[18-6-2020 17:35:53] Generating an outpost for the start of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 17:35:53] Generating an outpost for the end of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 17:35:43] Transitioning to Ino LineaAttis (current location: Attis City, selected location: Ino Linea Institute of Astrobiology, leaving sub: Typhon, at start: True, at end: False, transition type: LeaveLocation)
[18-6-2020 17:35:35] Generating an outpost for the start of the level... (Location: Attis City, level type: Outpost)
[18-6-2020 17:35:26] Transitioning to Attis (current location: Attis City, selected location: Ino Linea Institute of Astrobiology, leaving sub: Typhon, at start: True, at end: False, transition type: ReturnToPreviousLocation)
[18-6-2020 17:34:53] Generating an outpost for the start of the level... (Location: Ino Linea Institute of Astrobiology, level type: LocationConnection)
[18-6-2020 17:34:53] Generating an outpost for the end of the level... (Location: Attis City, level type: LocationConnection)
[18-6-2020 17:34:43] Transitioning to Ino LineaAttis (current location: Attis City, selected location: Ino Linea Institute of Astrobiology, leaving sub: Typhon, at start: True, at end: False, transition type: LeaveLocation)
[18-6-2020 17:28:15] Generating an outpost for the start of the level... (Location: Attis City, level type: Outpost)
[18-6-2020 17:25:38] OpenAL Soft on Microphone (Steam Streaming Microphone) 54
[18-6-2020 17:25:38] OpenAL Soft on Digital Audio Interface (Valve VR Radio & HMD Mic) 65
[18-6-2020 17:25:38] OpenAL Soft on Microphone (DroidCam Virtual Audio) 50
[18-6-2020 17:25:38] OpenAL Soft on Microfoon (4- USB PnP Sound Device) 50
[18-6-2020 17:25:32] Error in content package "BetterCharacterHead" - no content files defined.
[18-6-2020 17:25:32] Error in content package "BetterCharacterHead" - no content files defined.
[18-6-2020 17:25:19] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[18-6-2020 17:25:19] Content package load order: Vanilla 0.9
[18-6-2020 17:25:19] Error in content package "BetterCharacterHead" - no content files defined.
[18-6-2020 17:25:18] Error in content package "BetterCharacterHead" - no content files defined.
[18-6-2020 17:25:18] Error in content package "BetterCharacterHead" - no content files defined.
[18-6-2020 17:25:18] Content package load order: Vanilla 0.9
[18-6-2020 17:25:18] Content package load order: Vanilla 0.9
[18-6-2020 17:25:16] Attempting to open ALC device "OpenAL Soft"
[18-6-2020 17:25:15] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[18-6-2020 17:25:14] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[18-6-2020 17:25:14] Error in content package "BetterCharacterHead" - no content files defined.
```

**Comment by ekqvist** [comment]
Seem to be able to consistently reproduce this by spawning in a monster, killing it, and attempting to open the contextual command interface while focused on the dead monster

**Comment by ekqvist** [comment]
Should be fixed in Regalis11/Barotrauma-development@7b0b7fe. Added some extra null checks. Required the target character to be alive, friendly, and have the same team ID. Should do some testing by opening the command UI on different characters to see that everything's working as it should.

**Comment by Rokvach** [comment]
Wasn't able to reproduce the issue using the steps above, so seems to be working correctly. Closing.
